Source/data/iterators.hpp

```diff
@@ -1,234 +1,415 @@
+				*state_ = GetNextField(result.ptr, end_);
+				// and prepend what was already parsed
+				state_->value = { begin, (state_->value.data() - begin) + state_->value.size() };
+			}
+		} else {
+			result = std::from_chars(state_->value.data(), end_, destination);
 		}
+		return result.ec;
+	}
 
-		[[nodiscard]] bool operator!=(const Iterator &rhs) const
-		{
-			return !(*this == rhs);
+	template <typename T>
+	[[nodiscard]] tl::expected<T, std::errc> asInt()
+	{
+		T value = 0;
+		auto parseIntResult = parseInt(value);
+		if (parseIntResult == std::errc()) {
+			return value;
 		}
+		return tl::unexpected { parseIntResult };
+	}
 
-		/**
-		 * Advances to the next field in the current record
-		 */
-		Iterator &operator++()
-		{
-			return *this += 1;
+	/**
+	 * @brief Checks if this field is the last field in a file, not just in the record
+	 *
+	 * If you call this method before calling parseInt or operator* then this will trigger a
+	 * read, it's more efficient to call this after calling parseInt.
+	 * @return true if this field is the last field in the file/RecordsRange
+	 */
+	[[nodiscard]] bool endOfFile()
+	{
+		if (state_->status == GetFieldResult::Status::ReadyToRead) {
+			*state_ = GetNextField(state_->next, end_);
 		}
+		return state_->endOfFile();
+	}
 
-		/**
-		 * @brief Advances by the specified number of fields
-		 *
-		 * if a non-zero increment is provided and advancing the iterator causes it to reach the end
-		 * of the record the iterator is invalidated. It will compare equal to an end iterator and
-		 * cannot be used for value access or any further parsing
-		 * @param increment how many fields to advance (can be 0)
-		 * @return self-reference
-		 */
-		Iterator &operator+=(unsigned increment)
-		{
-			if (state_->status == GetFieldResult::Status::ReadyToRead)
-				*state_ = DiscardMultipleFields(state_->next, end_, increment);
-
-			if (state_->endOfRecord()) {
-				state_ = nullptr;
-			} else {
-				// We use Status::ReadyToRead as a marker so we only read the next value on the next call to operator*
-				// this allows consumers to read from the input stream without using operator* if they want
-				state_->status = GetFieldResult::Status::ReadyToRead;
-			}
+	[[nodiscard]] bool operator==(std::string_view rhs) const
+	{
+		return state_->value == rhs;
+	}
+
+	/**
+	 * Returns the current row number
+	 */
+	[[nodiscard]] unsigned row() const
+	{
+		return row_;
+	}
+
+	/**
+	 * Returns the current column/field number (from the start of the row/record)
+	 */
+	[[nodiscard]] unsigned column() const
+	{
+		return column_;
+	}
+
+	/**
+	* Allows accessing the value of this field in a const context
+	*
+	* This requires an actual non-const value access to happen first before it returns
+	* any useful results, intended for use in error reporting (or test output).
+	*/
+	[[nodiscard]] std::string_view currentValue() const
+	{
+		return state_->value;
+	}
+};
+
+#ifdef BUILD_TESTING

```

Comment:
This can be defined in the test itself or just always defined (LTO will remove unused code)

Reply:
Thanks, I'm definitely glad to remove the `#ifdef` guard :D (guess we should do the same for `PointOf` and friends as well)